File: solutions/r/leap/2/leap.R
leap <- function(year) {
  # is this year for divisible by 4?
  if (year %% 4 == 0) {
    # is this year divisible by 100?
    if (year %% 100 == 0) {
      # is this year divisible by 400?
      if (year %% 400 == 0) {
        # this year, divisible by 400, is a leap year
        return(TRUE)
      }
      # this year, divisible by 100, but not by 400, is not a leap year
      return(FALSE)
    }
    # this year, divisble by 4, but not by 100, is a leap year
    return(TRUE)
  }
  # this year, not divisible by 4, is not a leap year
  return(FALSE)
}

File: solutions/clojure/leap/1/src/leap.clj
(ns leap)

;; use the cond guard with mod function to determine if leap or not
(defn leap-year? "Determines if leap year or not" [year]
      (cond
        ;; divisible by 400, thus leap
        (zero? (mod year 400)) true
        ;; divisible by 100, but not 400, thus not leap
        (zero? (mod year 100)) false
        ;; divisible by 4, but not 100, thus leap
        (zero? (mod year 4)) true
        ;; not divisible by 4, thus not leap
        :else false))


